central/reports/service/v2/service_impl.go

```diff
@@ -0,0 +1,73 @@
+package v2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/stackrox/rox/central/reportconfigurations/service/common"
+	metadataDS "github.com/stackrox/rox/central/reports/metadata/datastore"
+	apiV2 "github.com/stackrox/rox/generated/api/v2"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/search"
+	"google.golang.org/grpc"
+)
+
+const maxPaginationLimit = 1000
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type serviceImpl struct {
+	metadataDatastore metadataDS.DataStore
+}
+
+func (s *serviceImpl) RegisterServiceServer(grpcServer *grpc.Server) {
+	if features.VulnMgmtReportingEnhancements.Enabled() {
+		apiV2.RegisterReportServiceServer(grpcServer, s)
+	}
+}
+
+func (s *serviceImpl) RegisterServiceHandler(ctx context.Context, mux *runtime.ServeMux, conn *grpc.ClientConn) error {
+	if features.VulnMgmtReportingEnhancements.Enabled() {
+		return apiV2.RegisterReportServiceHandler(ctx, mux, conn)
+	}
+	return nil
+}
+
+func (*serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
+	return ctx, common.Authorizer.Authorized(ctx, fullMethodName)
+}
+
+func (s *serviceImpl) GetReportStatus(ctx context.Context, req *apiV2.ResourceByID) (*apiV2.ReportStatus, error) {
+	if req == nil || req.GetId() == "" {
+		return nil, errors.New("Empty request or id")
+	}
+	rep, found, err := s.metadataDatastore.Get(ctx, req.GetId())
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, fmt.Errorf("Report not found for id %s", req.GetId())
```

Comment:
use: `errors.Errorf`

Reply:
fmt.Errorf is the right method. There is no errors.Errorf

Reply:
You have to use the package `"github.com/pkg/errors"`